core/src/trezor/wire/__init__.py

```diff
@@ -74,15 +75,27 @@
 workflow_packages = {}  # type: Dict[int, Tuple[str, str]]
 
 
+experimental_types = set()  # type: Set[int]
+experimental_enable = False  # type: bool
+
+
 def add(wire_type: int, pkgname: str, modname: str) -> None:
     """Shortcut for registering a dynamically-imported Protobuf workflow."""
     workflow_packages[wire_type] = (pkgname, modname)
 
+    msgtype = messages.get_type(wire_type)
```

Comment:
welp
this is not great
`get_type` causes an import, which we don't want because it might in practice eat our precious memory

i'm wondering if we could just register the message as normal, and only raise in `handle_session` where we call `get_type` anyway?

Reply:
Good point. What you suggest seems possible, will look into it.

Reply:
Pushed a059f38 which moves the check to `load_message` where field checks can also be added (e69c74f / cc @andrewkozlik).